Polish seed UI options

JohanAR opened the pull request:
Added a checkbox for random seed, instead using the "magic number" of -1 for this. Writing a valid number in either of the seed boxes automatically turns off random.

Moved variation seed to a row so that it is grouped with variation strength, since these work together.

Added boxes that display last seed, and last variation seed, with buttons that copy these to the seed/vseed inputs.

JohanAR committed:
Add buttons for random and reuse seed.

Random button sets seed to -1, reuse copies the seed from the last
generated image.

dfaker commented:
It's nice to be able to pass -1 as a seed into scripts such as "X/Y plot" for exploration:

![image](https://user-images.githubusercontent.com/35278260/190800061-350a4599-2a31-44a1-b21b-a28f518307fd.png)

So I'd not like to see special handling of -1 totally disappear.
 
What was your intent in making this? particularly as the real seed used is piped out below most of the runs.

AUTOMATIC1111 commented:
We talked on discord about this a bit.

Overall, I don't want a checkbox, I don't want extra args passing to the img2img/txt2img without necessity; a button is possible to just set seed field to -1, but it has to be nice, small, next to seed field and have an icon instead of text.

AUTOMATIC1111 commented:
will poland ever deliver

JohanAR commented:
*Sweden

Been busy playing with my new GPU, but I'll revert all the unnecessary stuff and fix it now :)

JohanAR commented:
> What was your intent in making this? particularly as the real seed used is piped out below most of the runs.

Just wanted to streamline my workflow, which often switches between randoming new seeds and keeping the last random when I want to further explore a result that I like. Thought it was annoying that it wasn't a single mouse click operation to switch between the two.

JohanAR commented:
There, finally re-implemented the logic, cleaned things up and rebased it. The variation strength slider needs to be vertically centered but I gave up fighting the css, maybe you know how to easily fix it..

And yes, I got the joke eventually.. Just needed my morning coffee :S

AUTOMATIC1111 commented:
in it goes

AUTOMATIC1111 commented:
so far, first random button does not have a tooltip and extra section is not hidden by default

AUTOMATIC1111 commented:
aha, visibility is fixed by gradio 3.3.1

AUTOMATIC1111 commented:
in more grim news, this broke a lot of things because you changed constructor for Processed but did not change its uses in most places.

AUTOMATIC1111 commented:
it was because you used the literal emoji character for label in the seed field, instead of the variable with character code you defined and used for subseed field

AUTOMATIC1111 commented:
the buttons are great, very useful

JohanAR commented:
> it was because you used the literal emoji character for label in the seed field, instead of the variable with character code you defined and used for subseed field

Oops, kind of embarrassing miss :/

Glad you like them! I also use them a lot now. Thanks for your support and patience
